From 242efff9961f10243c928a2c52719527534bf7a0 Mon Sep 17 00:00:00 2001
From: Chen Tianmi <tianmi.chen@intel.com>
Date: Tue, 30 Apr 2013 14:17:10 +0800
Subject: libstagefright: Ignore stsd box flag check for MPEG4Extractor

BZ: 50109

Ignore stsd box flag check to have more MP4 files played.

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Change-Id: Ib400345a837f34d3a497c380511b134700de3055
Orig-Change-Id: Icbe9af5c8b1b9f046b0b8ddda9b48341261f8fae
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 02137ce..b989013 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -1183,11 +1183,6 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
                 return ERROR_IO;
             }
 
-            if (U32_AT(buffer) != 0) {
-                // Should be version 0, flags 0.
-                return ERROR_MALFORMED;
-            }
-
             uint32_t entry_count = U32_AT(&buffer[4]);
 
             if (entry_count > 1) {
-- 
1.7.4.1

